pkg/pipeline/output/bin.go

```diff
@@ -80,32 +78,33 @@ func (b *Bin) Link() error {
 }
 
 func (b *Bin) linkSink(sink *streamSink) error {
-	if b.isWebSource {
-		sinkPad := sink.sink.GetStaticPad("sink")
-
-		// intercept FlowFlushing
-		proxy := gst.NewGhostPad("proxy", sinkPad)
-		proxy.SetChainFunction(func(self *gst.Pad, _ *gst.Object, buffer *gst.Buffer) gst.FlowReturn {
-			internal, _ := self.GetInternalLinks()
-			if len(internal) == 0 {
-				// there should always be exactly one
-				return gst.FlowNotLinked
-			}
-
-			flow := internal[0].Push(buffer)
-			if flow == gst.FlowFlushing {
-				return gst.FlowOK
-			}
-			return flow
-		})
-		proxy.ActivateMode(gst.PadModePush, true)
-
-		// link
-		if linkReturn := sink.queue.GetStaticPad("src").Link(proxy.Pad); linkReturn != gst.PadLinkOK {
-			return errors.ErrPadLinkFailed("rtmp sink", linkReturn.String())
+	sinkPad := sink.sink.GetStaticPad("sink")
+
+	// intercept FlowFlushing
+	proxy := gst.NewGhostPad("proxy", sinkPad)
+	proxy.Ref()
+
+	proxy.SetChainFunction(func(self *gst.Pad, _ *gst.Object, buffer *gst.Buffer) gst.FlowReturn {
+		buffer.Ref()
```

Comment:
do we need to explicitly deref these anywhere?

Reply:
Go gst buffers are instantiated with a `runtime.SetFinalizer(wrapped, (*Buffer).Unref)` call before they're returned